rfcs/rfc-net-20230821-1-headless-server-launcher.md

```diff
@@ -0,0 +1,128 @@
+
+*   [Application](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Application/Application.h#L44)
+*   [InputDeviceGamepad](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad.h#L28)
+*   [InputDeviceKeyboard](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h#L30)
+*   [InputDeviceMotion](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Input/Devices/Motion/InputDeviceMotion.h#L25)
+*   [InputDeviceMouse](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Input/Devices/Mouse/InputDeviceMouse.h#L27)
+*   [InputDeviceTouch](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Input/Devices/Touch/InputDeviceTouch.h#L23)
+*   [InputDeviceVirtualKeyboard](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Input/Devices/VirtualKeyboard/InputDeviceVirtualKeyboard.h#L22)
+*   [NativeWindow](https://github.com/o3de/o3de/blob/2e3f76beb03a976d4e4fcd819349506bbefc5746/Code/Framework/AzFramework/AzFramework/Windowing/NativeWindow.h#L95)
+
+These classes were designed to provide platform-specific services to the host machine through the use of the PIMPL idiom and Platform Abstraction Layer (PAL) pattern in O3DE. However, since `AzFramework` is a base library for O3DE and has a build dependency to almost all targets in O3DE, it will pull in dependencies to any platform-specific native libraries that handle windowing and input as well. This forces all projects to pull in these dependencies. On Linux platforms, that means it will pull in dependencies to XCB and X11 libraries, even if it does not use them.
+
+What is needed is to split off the platform-specific services and their dependent library dependencies from `AzFramework` into its own library,  `AzFramework.NativeUI`.  The abstraction of the windowing and input devices makes this possible. The PIMPL idiom allows us to move the platform-specific implementations of the native window and input device interactions with the platform into the new library, and the inclusion of that library can be specified in targets that intend to use them. In the case of the headless server launcher, it will be intentionally omitted. 
+
+When omitted, the opaque pointers to the underlying PIMPL implementations will be `null` , and the abstraction layers that hosts these PIMPLs will be protected with a `null` check and perform a no-op in those cases. This will in effect disable any attempt to read or write to the native window or devices in headless mode (if any).
```

Comment:
```suggestion
When omitted, the opaque pointers to the underlying PIMPL implementations will be `null`, and the abstraction layers that hosts these PIMPLs will be protected with a `null` check and perform a no-op in those cases. This will in effect disable any attempt to read or write to the native window or devices in headless mode (if any).
```